ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUpdaterThread.java

```diff
@@ -210,27 +213,34 @@ private void stopWorkers() {
     }
   }
 
-  private List<AnalyzeWork> processOneTable(TableName fullTableName)
+  private List<AnalyzeWork> processOneTable(TableName fullTableName, Map<String, Boolean> dbsToSkip)
       throws MetaException, NoSuchTxnException, NoSuchObjectException {
     if (isAnalyzeTableInProgress(fullTableName)) return null;
     String cat = fullTableName.getCat(), db = fullTableName.getDb(), tbl = fullTableName.getTable();
+    String dbName = MetaStoreUtils.prependCatalogToDbName(cat,db, conf);
+    if (!dbsToSkip.containsKey(dbName)) {
+      Database database = rs.getDatabase(cat, db);
+      boolean skipDb = false;
+      if (MetaStoreUtils.isDbBeingFailedOver(database)) {
+        skipDb = true;
+        LOG.info("Skipping all the tables which belong to database: {} as it is being failed over", db);
+      } else if (ReplUtils.isTargetOfReplication(database)) {
```

Comment:
There is lot of code duplication between this and PartitionManagement. Can we make not achieve by having a single copy?
Also,  why do we have two methods for isTargetOfReplication(), can we have just one?

Reply:
We already had two separate methods declared in both ReplUtils and PartitionManagementTask because ReplUtils package is not accessible in MetastoreThreads. But it can be moved to MetastoreUtils where it can be accessible by both of them.